### PR TITLE
oc-mirror configurability

### PR DESCRIPTION
**Pull request by geoberle**
### What this PR does

support configuration via environment variables instead of config file. this way deployment via azure container app should be more convenient

Jira: 
Link to demo recording:

### CODE REVIEW

File: image-sync/oc-mirror/main.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package main
```
Comment (janboll) [inline image-sync/oc-mirror/main.go:1]: Nit: I not have used golang for this, but plain `.sh`, since it's available as well. :man_shrugging:

Comment (janboll) [reply image-sync/oc-mirror/main.go:1]: What about an alternative approach:

1. Use tempaltize to generate the configuration
2. Encode it base64, pass it in as environment variable
3. Decode and write it to a file

Comment (geoberle) [reply image-sync/oc-mirror/main.go:1]: i like the base64 option. i'll go for it

File: image-sync/deployment/Makefile [hunk 2]
```diff
@@ -25,11 +25,12 @@ deploy-oc-mirror:
 	--set serviceAccountName=image-sync \
 	--set azureClientId=$${AZURE_SYNC_MI_CLIENT_ID} \
 	--set acrRegistryName=${ARO_HCP_IMAGE_ACR} \
+	--set stableVersions="4.16\,4.17" \
 	--set azureTenantId=$${TENANT_ID} \
 	--set ocmirrorImage="arohcpdev.azurecr.io/image-sync/ocmirror" \
 	--set ocmirrorTag=latest \
 	--set credsPullSecret=pull-secret \
-	--set credsKeyVaultName=service-kv-aro-hcp-dev
+	--set credsKeyVaultName=aro-hcp-dev-svc-kv
```
Comment (bennerv) [inline image-sync/deployment/Makefile:33]: should this be a param?

File: image-sync/oc-mirror/Dockerfile [hunk 3]
```diff
@@ -1,9 +1,9 @@
-FROM mcr.microsoft.com/cbl-mariner/base/core:2.0 AS downloader
+FROM --platform=linux/amd64 mcr.microsoft.com/cbl-mariner/base/core:2.0 AS downloader
```
Comment (bennerv) [inline image-sync/oc-mirror/Dockerfile:1]: Do we need to specify platform here?  Should the container infer it depending on the platform you run on?

File: image-sync/oc-mirror/Makefile [hunk 4]
```diff
@@ -9,7 +9,6 @@ OC_MIRROR_IMAGE_TAGGED ?= $(OC_MIRROR_IMAGE):$(COMMIT)
 build-push: image push
 
 image:
-	cp ../configuration/mvp-oc-mirror.yml config.yml
 	docker build --platform="linux/amd64" -f "./Dockerfile" -t ${OC_MIRROR_IMAGE_TAGGED} .
```
Comment (bennerv) [inline image-sync/oc-mirror/Makefile:12]: Docs say "podman" can we update `docker` -> `podman` or vice versa?  Or do we have a preference?

File: image-sync/oc-mirror/imageset-config.yml.tmpl [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+apiVersion: mirror.openshift.io/v1alpha2
+storageConfig:
+  registry:
+    imageURL: {{ .RegistryUrl }}/mirror/oc-mirror-metadata
+    skipTLS: false
+mirror:
+  platform:
+    architectures:
+      - multi
+      - amd64
```
Comment (bennerv) [inline image-sync/oc-mirror/imageset-config.yml.tmpl:11]: Do we need amd64 images still, or can we switch everything to multi yet?

**Comment by geoberle** [comment]
closing this as it has been implemented as discussed in the #762

**Pull request closed without merging**